### ISSUE TITLE
[RFC] How to handle "error" and warning flashes on fresh sites. (was "We shouldn't display "errors" about missing `mailoptions` config on a fresh setup.")

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#4712]
Newcomers will think they've done something wrong. We should only display this when needed. (like in Bolt 2, I think)

![screenshot 2016-01-17 13 39 52](https://cloud.githubusercontent.com/assets/1833361/12377478/1b0b21da-bd20-11e5-94bc-7b332a595cda.png)

**Comment by GawainLynch** [comment]
Should we make it an information colour?

Worth noting, that is now only shown to users who have the access rights to edit `config.yml` so it does narrow the scope a bit. 

Also the message is a helpful one, as a "newer" admin might not know about the `mailoptions:` and lose a lot time trying to deal with users that can't reset passwords… which IMHO is a bigger liability for a site admin.

**Comment by bobdenotter** [comment]
> Worth noting, that is now only shown to users who have the access rights to edit config.yml so it does narrow the scope a bit.

Which will be _everyone_ who's setting up Bolt for the first time. For the sake of "making a good first impression", it should not be shown *at all* on new installs. 

I'd be happy with it, if it started showing up after a week or so.

**Comment by GawainLynch** [comment]
>> Worth noting, that is now only shown to users who have the access rights to edit `config.yml` so it does narrow the scope a bit.

> Which will be _everyone_ who's setting up Bolt for the first time. For the sake of "making a good first impression", it should not be shown *at all* on new installs. 

Like…

```
There are no users in the database. Please create the first user.
```

```
Note: You are currently using SQLite to set up the first user. If you wish to use 
MySQL or PostgreSQL instead, edit the configuration file at 
'app/config/config.yml' and Bolt will set up the database tables for you. Be sure 
to reload this page before continuing.
```

```
The database needs to be updated/repaired. Go to 'Configuration' > 'Check Database' 
to do this now.
```

```
Welcome to your new Bolt site, Gawain Lynch.
```

```
It seems there's no content in the database. To get started quickly, add some Lorem 
Ipsum dummy content.
```

```
You've been logged on successfully.
```

Which leads me back to my first question, would it be better to go with an "information" colour so that people don't feel they've done something wrong, but are rather getting *helpful* hints towards tasks that *should* be completed for their Bolt site to function seamlessly :wink: 

> I'd be happy with it, if it started showing up after a week or so.

As above though. Also what about when someone sets up a new site, forgets their password 5 minutes later, and can't figure out *why* they're not getting reset notifications.

**Comment by bobdenotter** [comment]
Well, I do think there are too much of those notices anyways. We'll need to think of a way to reduce those. Sometime you'll get three or even four stacked on top. Less than optimal.

**Comment by GawainLynch** [comment]
Yeah, agreed… I had stayed away from touching it as I thought they were "little darlings", but I am guessing we're on the same page… as usual :grin: 

What seems to me to be the obvious solution is a service, that gets saved in session… but that is feature work and… we have a really bad habit around those and late freezes.

**Comment by jadwigo** [comment]
for inspiration: have you guys seen the drupal system status page? It has a nice overview of what's wrong, or correctly setup on your site.
![screenshot-selection_123](https://cloud.githubusercontent.com/assets/1488214/12395294/14afa0b8-be01-11e5-9b48-e3d830586ec2.png)

We might have a similar page next to the database checks and the system log where the mailoptions and other status messages will be shown by default.

To alert a user when multiple informational and warning messages are active we could display a frienldly message like "There are some additional settings you can to change for your site. See the status page for more information.".

To prevent multiple useless clicks if only one of the messages is active we can just show that message.

Also I agree the mailconfiguration would be better as an informational message instead of a warning message.

**Comment by GawainLynch** [comment]
I hadn't, but I did write the [Bolt\Configuration\Check](https://github.com/bolt/bolt/tree/master/src/Configuration/Check) classes early last year, but asked for a UI person to help with the, um, UI part.

I think a combination of all of this would be good. I wasn't overly a fan of the original approach as putting the call into the extension path is an anti-pattern, and doesn't address the core problem… people miss stuff and that can bite you with things like password resets.

**Comment by SahAssar** [comment]
Perhaps having some of these in a checklist in the right sidebar? We could give it some friendlier title like "Before opening your site to the public we recommend you have a look at these:". It would give users a sense that they are going towards a goal as they see stuff getting checked off and we could link to relevant sections in the docs/manual.

For seasoned users it could be disabled in config.yml just like news and the stack can be now.

**Comment by GawainLynch** [comment]
Abso… it just becomes a question of how/when/who wrt to implementation.

**Comment by SahAssar** [comment]
Sorry, but should this really be "blocking release"? I'm removing the label for now, but if anyone really feels like this is truly blocking please add it back.

**Comment by jadwigo** [comment]
It's technically not blocking the release because we will have a fully functional cms without a solution to this.

But as @bobdenotter already stated its the first impression that new implementers who are setting up a site get. A new user can interpret it as an "OMG I messed up something" signal - which sends the wrong message.

Also UI/UX issues like this should be important enough to warrant release blockers - because it's stuff that 2 out of three groups of our core audience have to deal with.

**Comment by GawainLynch** [comment]
Yeah, what @jadwigo said.

@bobdenotter I am wondering if we could actually solve this with a widget in the side bar, with something like this:

![image](https://cloud.githubusercontent.com/assets/1427081/12940059/bc5f380c-cfc5-11e5-9317-2ffea465f291.png)

**Comment by bobdenotter** [comment]
@GawainLynch Absolutely. I'd prefer to make it look slightly different, but that'd be a good approach. I'll take this one.

**Comment by GawainLynch** [comment]
Yeah, that's just a 10 second mock up (that crashed Chrome twice) from the person who can't even draw stick figures… let's not get me started on UI full stop :laughing:

**Comment by bobdenotter** [comment]
Another one to add a check for: "Fatal error: Maximum function nesting level of '256' reached, aborting! "

**Comment by bobdenotter** [comment]
Fixed by #4881.
